### ISSUE TITLE
CMake ignores virtualenv python (Trac #1463)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1455]
If you invoke cmake from within a virtualenv on OS X, you'll get the system /usr/bin/python no matter what. It looks like this is known behavior:

https://cmake.org/pipermail/cmake/2014-August/058368.html

We should probably prefer whatever is first in PATH as the linked post suggests. This has the potential to remove a broad class of gotchas in which icetray is linked against the system python but loaded from a different python at runtime. Thoughts?

Migrated from https://code.icecube.wisc.edu/ticket/1463
```json
{
    "status": "closed", 
    "changetime": "2019-01-12T00:05:09", 
    "description": "If you invoke cmake from within a virtualenv on OS X, you'll get the system /usr/bin/python no matter what. It looks like this is known behavior:\n\nhttps://cmake.org/pipermail/cmake/2014-August/058368.html\n\nWe should probably prefer whatever is first in PATH as the linked post suggests. This has the potential to remove a broad class of gotchas in which icetray is linked against the system python but loaded from a different python at runtime. Thoughts?", 
    "reporter": "jvansanten", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1547251509451813", 
    "component": "cmake", 
    "summary": "CMake ignores virtualenv python", 
    "priority": "normal", 
    "keywords": "", 
    "time": "2015-12-03T13:42:09", 
    "milestone": "", 
    "owner": "nega", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20151203T14:14:00: kjmeagher commented:

> This prevents datio-pyshovel from being build unless one installs `urwid` as root

**Comment by icecube-trac** [comment]
Trac update at 20151203T15:01:34: 
* nega changed resolution from "" to "fixed"
* nega changed owner from "" to "nega"
* nega commented:

> In changeset 2355:
> 
> ```text
> #!CommitTicketReference repository="IceTray" revision="2355"
> use python in the PATH
> 
> fixes #1455
> ```


* nega changed status from "new" to "closed"

**Comment by icecube-trac** [comment]
Trac update at 20190112T00:05:09: nwhitehorn commented:

> In changeset 170151:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="170151"
> use python in the PATH
> 
> fixes #1455
> ```
